### PR TITLE
Bluetooth: controller: Fix rejected enc procedure not terminated

### PR DESCRIPTION
**Pull request by joerchan**
Fix issue in the handling of LL_REJECT_EXT_IND packets, this would look
at the procedures that are enqueued, and not the procedure that was
being rejected. This meant that although a reject was received for the
encryption procedure, the handling for a different control procedure was
run.
This would result in the link being terminated as control procedure
timer would time out for the encryption procedure.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ctrl.c [hunk 1]
```diff
@@ -2277,55 +2277,60 @@ isr_rx_conn_pkt_ctrl_rej_phy_upd(struct radio_pdu_node_rx *node_rx,
...
+		*rx_enqueue = 1U;
+		break;
+#endif /* CONFIG_BT_CTLR_LE_ENC */
 
-			/* Procedure complete */
-			_radio.conn_curr->llcp_ack = _radio.conn_curr->llcp_req;
-			_radio.conn_curr->procedure_expire = 0U;
+#if defined(CONFIG_BT_CTLR_PHY)
+	case PDU_DATA_LLCTRL_TYPE_PHY_REQ:
+		if (_radio.conn_curr->llcp_phy.ack !=
```
Comment (joerchan) [inline subsys/bluetooth/controller/ll_sw/ctrl.c]: @cvinayak I think these check should all be llcp_ack != llcp_req && llcp_type != PHY/CONN_PARAM/LENGTH, right?
Could also add one for the ENC procedure?

File: subsys/bluetooth/controller/ll_sw/ctrl.c [hunk 2]
```diff
@@ -2277,55 +2277,60 @@ isr_rx_conn_pkt_ctrl_rej_phy_upd(struct radio_pdu_node_rx *node_rx,
 static inline void
 isr_rx_conn_pkt_ctrl_rej(struct radio_pdu_node_rx *node_rx, u8_t *rx_enqueue)
```
Comment (joerchan) [inline subsys/bluetooth/controller/ll_sw/ctrl.c]: @cvinayak There is similar code in the SPLIT controller. But I think this bug is a product of the parallel procedure handling. So I'm not certain if the fix should be applied there yet.

File: subsys/bluetooth/controller/ll_sw/ctrl.c (hunk 1 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ctrl.c]: Please add the check for ENC procedure. For others with state value, I am assuming individual checks are needed (I can rework in a new PR).

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/ctrl.c (hunk 1 above)
Comment (joerchan) [reply subsys/bluetooth/controller/ll_sw/ctrl.c]: Added check for the ENC procedure. The other ones I'm not sure about. So I'll leave them as is.

File: subsys/bluetooth/controller/ll_sw/ctrl.c [hunk 3]
```diff
@@ -2274,58 +2274,76 @@ isr_rx_conn_pkt_ctrl_rej_phy_upd(struct radio_pdu_node_rx *node_rx,
...
-	} else {
-		struct pdu_data_llctrl_reject_ext_ind *rej_ext_ind;
-		struct pdu_data *pdu_rx;
+	/* Procedure complete */
+	_radio.conn_curr->llcp_ack = _radio.conn_curr->llcp_req;
+	_radio.conn_curr->procedure_expire = 0U;
 
-		pdu_rx = (void *)node_rx->pdu_data;
-		rej_ext_ind = (void *)&pdu_rx->llctrl.reject_ext_ind;
+	*rx_enqueue = 1U;
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ctrl.c]: Move this to after isr_rx_conn_pkt_ctrl_rej_enc call.

File: subsys/bluetooth/controller/ll_sw/ctrl.c [hunk 4]
```diff
@@ -2274,58 +2274,76 @@ isr_rx_conn_pkt_ctrl_rej_phy_upd(struct radio_pdu_node_rx *node_rx,
...
+	case PDU_DATA_LLCTRL_TYPE_ENC_REQ:
+		if (_radio.conn_curr->llcp_ack != _radio.conn_curr->llcp_req &&
+		    _radio.conn_curr->llcp_type == LLCP_ENCRYPTION) {
 			/* enqueue as if it were a reject ind */
 			pdu_rx->llctrl.opcode = PDU_DATA_LLCTRL_TYPE_REJECT_IND;
 			pdu_rx->llctrl.reject_ind.error_code =
-				rej_ext_ind->error_code;
-			*rx_enqueue = 1U;
-			break;
+			rej_ext_ind->error_code;
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ctrl.c]: Move these below isr_rx_conn_pkt_ctrl_rej_enc

**Review by cvinayak — CHANGES_REQUESTED** [review]
LGTM

File: subsys/bluetooth/controller/ll_sw/ctrl.c (hunk 3 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ctrl.c]: the function is reused, hence this ok. resolved.

**Review by cvinayak — APPROVED** [review]
LGTM

File: subsys/bluetooth/controller/ll_sw/ctrl.c (hunk 4 above)
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ctrl.c]: missing indentation

**Review by cvinayak — APPROVED** [review]
LGTM

**Pull request merged**